pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -220,6 +159,112 @@ def run_func(self, *args, **kwargs):
 
         return decorator
 
+    def _get_pattern_lst(self, source_path, timestamps=False):
+        '''
+        Get an order list follow the name pattern
+        '''
+        if timestamps:
+            pattern, ext = os.path.basename(
+                os.path.normpath(source_path)).split(".")
```

Comment:
I would highly recommend to use https://docs.python.org/3/library/os.path.html#os.path.splitext to split extensions.

Reply:
Yes, you are right.